CONTRIBUTING.md

```diff
@@ -1,43 +1,95 @@
 
-### Item Macros
+### Data layout
+The `module/data/` directory is laid out as follows:
+- Each 'datatype' (the array names in 5eTools' JSON format—`"monster"`, `"spell"`, `"classFeature"`, etc.) is given its own directory.
+- Within that directory, a `__core.json` file contains data for all entities which are natively available on 5eTools (i.e. without loading homebrew).
+- Within the same directory, each homebrew source has its own file named `<brewSourceJson>.json` (e.g. `WJMAiS.json` for *Wildjammer: More Adventures in Space*). If a homebrew source has multiple datatypes, one file per datatype is required (excluding datatypes without automation).
 
-Item macros are stored as JavaScript files, and built into the data when the module is packaged. The source for these macros can be found in the [macro-item](./macro-item) directory. **Note that the first and last lines of the macro are stripped when the macro is compiled into the module's data.**
+### Macros
+Many facets of automation must be handled with macros. These are attached using [Item Macro](https://foundryvtt.com/packages/itemacro/), which appear as an escaped string in data. To add some human-readability, macros are handled separately and merged during the module build.
 
-A new macro can be created using the following command:
+In the [`macro-item/`](./macro-item) directory is a directory for each datatype. Save your (well-formatted, commented) macro code as a Javascript file in one of these directories, structured as an async function named `macro`, with filename `<sourceJson>_<item-name-lowercase-hyphenated>.js` (e.g. `XGE_toll-the-dead.js`). **Note that the first and last lines of the file**—the ones that turn the macro into an async function—**are stripped on compilation into the module's data**.
 
+You can create a new macro file using the `npm run mt --` command. The directory (`-d`) should match that of the JSON file into which the macro will be built.
 ```bash
-# ex:
-# npm run mt -- -d spell -s PHB -n "Fireball"
-npm run mt -- -d <directory> -s <jsonSource> -n <entityName>
+npm run mt -- -d <directory> -s <jsonSource> -n <entity name>
+# example: npm run mt -- -d spell -s PHB -n "toll the dead"
+# example: npm run mt -- -d feat -s PHB -n "war caster"
 ```
 
-The directory (`-d`) should match that of the JSON file into which the macro will be built. The macro can then be referenced from within a JSON file by using: `"itemMacro": "<filename>"` (`"itemMacro": "PHB_fireball.js"` in the example given above)
+To reference this human-readable macro in the main data files, use `"itemMacro": "<filename>"` (`"itemMacro": "XGE_toll-the-dead.js"` in the example given above—see [`module/data/spell/__core.json`](https://github.com/TheGiddyLimit/plutonium-addon-automation/blob/master/module/data/spell/__core.json#L224)).
```

Comment:
linking `master` is a bit dodgy, maybe just inline the example here?

Reply:
The `master` link was just to see it in context (since I personally and presumably others find that more useful). But since there really isn't any context to be provided without just presenting the entire spell itself—in which case the reader can just see the `__core.json` file themselves!—I'll just delete the link and keep it simple.